tests/tools/utils.py

```diff
@@ -0,0 +1,99 @@
+        self._s3 = boto3.client(
+            's3',
+            config=botocore.client.Config(signature_version=botocore.UNSIGNED)
+        )
+        if download_location is None:
+            path = os.path.join(
+                os.path.dirname(__file__),
+                "../../resources")
+            self._default_download_location = os.path.abspath(path)
+        else:
+            self._default_download_location = download_location
+
+        print(
+            "Setting default download location: ", self._default_download_location,
+            file=sys.stderr)
+
+    def download(self, resource_type, resource_name, tags={}, version=None, download_location=None):
+        """Downloads the resource specified by name and type. Optionally, the user can
+        specify what tags the resource needs to have, and its version.
+
+        The default version used when downloading is the latest version as defined in the
+        resource manifest."""
+        # TODO: The resource_name does not need to be required, because we can download
+        # resources by type & tags.
+        if version is None:
+            version = self._latest_version
+        if download_location is None:
+            download_location = self._default_download_location
+        if version in self._resource_manifest.keys():
+            resource_parent = self._resource_manifest[version]
+        else:
+            raise Exception("Invalid version: {}".format(version))
+
+        resources = [r for r in resource_parent
+                     if r["resource_type"] == resource_type
+                     and r["resource_name"] == resource_name
+                     and resource_has_tags(r, tags)]
+        if len(resources) == 0:
+            print("No resources found", file=sys.stderr)
+            exit(1)
+
+        downloaded_file_paths = []
+        for resource in resources:
+            abs_dir = os.path.join(download_location, resource["relative_path"])
+            print(
+                "Creating download location if it does not exist: ", abs_dir,
+                file=sys.stderr)
+            os.makedirs(abs_dir, exist_ok=True)
+            abs_path = os.path.join(abs_dir, resource_name)
+
+            object_key = "{}/{}/{}".format(version, resource_type, resource_name)
+            print("Object to download from S3:", object_key, file=sys.stderr)
+            print("Downloading file to: ", abs_path, file=sys.stderr)
+
+            if not os.path.exists(abs_path):
+                self._s3.download_file(self._resource_bucket, object_key, abs_path)
+            downloaded_file_paths.append(abs_path)
+
+        return downloaded_file_paths
+
+    def get_latest_version(self):
+        """Get the latest version as defined in the resource manifest file."""
+        version_re = re.compile("^v[{0-9}]+")
+        versions = [key for key in self._resource_manifest.keys() if version_re.match(key)]
+        return "v{}".format(max(map(lambda v: v[1:], versions)))
```

Comment:
Note: This will be lexicographical comparison so "10" < "9", So something like `sorted(versions, key=int, ...)[-1]` or `map(lambda v: int(v[1:])...` is better and since we control the json better to get rid of the `v` in `v1` in the `json` itself?